### ISSUE TITLE
Spread adoption to your favorite SDKs

### ISSUE THREAD

**Issue by nicolas-grekas** [FriendsOfPHP/well-known-implementations#2]
As stated in the README:

> If your favorite SDK does not use this package yet, please let them know about it or better: send them a PR!

I'm creating this issue to keep track of such initiatives.
Please link back and/or drop a comment here when applicable.

See e.g.
https://github.com/composer/composer/issues/6758
https://github.com/getsentry/sentry-php/discussions/1411
https://github.com/async-aws/aws/issues/1318
 
Packages dependent on supported abstractions:
- https://packagist.org/packages/php-http/async-client-implementation/dependents?order_by=downloads
- https://packagist.org/packages/php-http/client-implementation/dependents?order_by=downloads
- https://packagist.org/packages/psr/http-client-implementation/dependents?order_by=downloads
- https://packagist.org/packages/psr/http-factory-implementation/dependents?order_by=downloads
- https://packagist.org/packages/psr/http-message-implementation/dependents?order_by=downloads

See also https://packagist.org/packages/php-http/discovery/dependents?order_by=downloads

**Comment by ostrolucky** [comment]
https://github.com/aws/aws-sdk-php could be a big win

**Comment by nicolas-grekas** [comment]
Some other ideas:
- https://packagist.org/packages/akeneo/api-php-client
- https://packagist.org/packages/jolicode/slack-php-api
- https://packagist.org/packages/vonage/client
- https://packagist.org/packages/m4tthumphrey/php-gitlab-api
- https://packagist.org/packages/knplabs/github-api
- https://packagist.org/packages/auth0/auth0-php
- https://packagist.org/packages/payum/payum
- https://packagist.org/packages/geocoder-php/common-http
- https://packagist.org/packages/omnipay/common
- https://packagist.org/packages/intercom/intercom-php
- https://github.com/microsoftgraph/msgraph-sdk-php-core
- https://packagist.org/packages/league/oauth2-client
- https://packagist.org/packages/league/oauth1-client

**Comment by nicolas-grekas** [comment]
> [aws/aws-sdk-php](https://github.com/aws/aws-sdk-php) could be a big win

I spent a few hours on this and my conclusion is that `aws-sdk-php` is too coupled to Guzzle: many things are configured via request options and those are Guzzle-specific. It would be possible to map them to Symfony HttpClient's option, but that wouldn't solve anything related to this discussion since here we're talking about coding against abstractions (PSR-18 / HTTPlug), and none of these abstractions support said options (excerpt: sink, http_stats_receiver, decode_content, stream, delay, headers.Accept-Encoding, curl.CURLOPT_TCP_KEEPALIVE, debug, proxy, verify, timeout, connect_timeout. etc.)

TL;DR: use https://github.com/async-aws/aws instead.

**Comment by nicolas-grekas** [comment]
Closing in favor of https://github.com/php-http/discovery/pull/208
